### PR TITLE
Feature/hw 58097 v3 python sdk for transfer refunds

### PR DESCRIPTION
**Pull request by madhankumar3103**
Changes has been done for transfer refunds

### CODE REVIEW

File: hyperwallet/tests/test_api.py [hunk 1]
```diff
@@ -2189,6 +2192,80 @@ def test_list_webhooks_success(self, mock_get):
 
         self.assertTrue(response[0].token, self.data.get('token'))
 
+    '''
+
+    Upload Documents
+
+    '''
+    def test_uploadDocumentsForUser_fail_need_user_token(self):
```
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2200]: It looks like that you added this method. Pls check. this doesn't look related to transfer refund.

File: hyperwallet/tests/test_api.py [hunk 2]
```diff
@@ -2189,6 +2192,80 @@ def test_list_webhooks_success(self, mock_get):
...
+
+        with self.assertRaises(HyperwalletException) as exc:
+            self.api.uploadDocumentsForUser()
+
+        self.assertEqual(exc.exception.message, 'userToken is required')
+
+    @mock.patch('hyperwallet.utils.ApiClient._makeRequest')
+    def test_uploadDocumentsForUser_success(self, mock_put):
```
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2208]: It looks like that you added this method. Pls check. this doesn't look related to transfer refund.

File: hyperwallet/tests/test_api.py [hunk 3]
```diff
@@ -2189,6 +2192,80 @@ def test_list_webhooks_success(self, mock_get):
...
+        self.assertTrue(response.token, self.data.get('token'))
+
+    '''
+
+    Transfer Refunds
+
+    '''
+
+    @mock.patch('hyperwallet.utils.ApiClient._makeRequest')
+    def test_transfer_refunds_success(self, mock_post):
```
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2222]: Please follow the convention of method names. This should be test_create_transfer_refund_successful. Dont forget to have the type of operation(Create/get/list/update) mentioned in the method's name.

File: hyperwallet/tests/test_api.py (hunk 1 above)
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2200]: It looks like that you added this method. Pls check. this doesn't look related to transfer refund.

File: hyperwallet/tests/test_api.py [hunk 4]
```diff
@@ -2189,6 +2192,80 @@ def test_list_webhooks_success(self, mock_get):
...
+
+        mock_post.return_value = self.data
+        response = self.api.createTransferRefund('token',self.data)
+        self.assertTrue(response.token, self.data.get('token'))
+
+    def test_transfer_refunds_fail_need_payment_token(self):
```
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2228]: Please follow the convention of method names. This should be test_create_transfer_refund.... Dont forget to have the type of operation(Create/get/list/update) mentioned in the method's name.

File: hyperwallet/tests/test_api.py [hunk 5]
```diff
@@ -2189,6 +2192,80 @@ def test_list_webhooks_success(self, mock_get):
...
+
+        with self.assertRaises(HyperwalletException) as exc:
+            self.api.createTransferRefund()
+
+        self.assertEqual(exc.exception.message, 'transferToken is required')
+
+    def test_transfer_refunds_fail_need_data(self):
```
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2235]: Please follow the convention of method names. This should be test_create_transfer_refunds.... Dont forget to have the type of operation(Create/get/list/update) mentioned in the method's name.

File: hyperwallet/tests/test_api.py [hunk 6]
```diff
@@ -2189,6 +2192,80 @@ def test_list_webhooks_success(self, mock_get):
...
+
+        self.assertEqual(exc.exception.message, 'data is required')
+
+    '''
+
+    Transfer Spend Back Refunds
+
+    '''
+
+    def test_transfer_spend_back_refunds_user_token(self):
```
Comment (Rrathinasabapath) [inline hyperwallet/tests/test_api.py:2248]: Please follow the convention of method names. This should be test_create_transfer_spend.... Dont forget to have the type of operation(Create/get/list/update) mentioned in the method's name.

File: hyperwallet/api.py [hunk 7]
```diff
@@ -2684,3 +2685,110 @@ def listWebhookNotifications(self,
 
     def __buildUrl(self, *paths):
         return '/'.join(s.strip('/') for s in paths)
+
+    def uploadDocumentsForUser(self,
```
Comment (Rrathinasabapath) [inline hyperwallet/api.py:2689]: It looks like you have added this method. Pls check if there is any code duplication and find out why its appearing so.

File: hyperwallet/utils/apiclient.py [hunk 8]
```diff
@@ -220,3 +222,25 @@ def __getRequestData(self, data):
         '''
 
         return (data if data is None else self.encryption.encrypt(data)) if self.encrypted else data
+
+    def putDocument(self, partialUrl, data, files):
```
Comment (Rrathinasabapath) [inline hyperwallet/utils/apiclient.py:226]: Please verify if the method has been added by you or if there is any duplication.

**Review by Rrathinasabapath — CHANGES_REQUESTED** [review]
Please make necessary changes

**Comment by arao6** [comment]
Please change the "branch to merge into" to V3 (not master).

File: hyperwallet/tests/test_api.py (hunk 3 above)
Comment (madhankumar3103) [reply hyperwallet/tests/test_api.py:2222]: method name has been changed.

File: hyperwallet/tests/test_api.py (hunk 5 above)
Comment (madhankumar3103) [reply hyperwallet/tests/test_api.py:2235]: method name has been changed.

File: hyperwallet/tests/test_api.py (hunk 4 above)
Comment (madhankumar3103) [reply hyperwallet/tests/test_api.py:2228]: method name has been changed.

File: hyperwallet/tests/test_api.py (hunk 6 above)
Comment (madhankumar3103) [reply hyperwallet/tests/test_api.py:2248]: method name has been changed.

File: hyperwallet/tests/test_api.py (hunk 1 above)
Comment (madhankumar3103) [reply hyperwallet/tests/test_api.py]: v3 merged code

File: hyperwallet/api.py (hunk 7 above)
Comment (madhankumar3103) [reply hyperwallet/api.py]: v3 merged code

File: hyperwallet/utils/apiclient.py (hunk 8 above)
Comment (madhankumar3103) [reply hyperwallet/utils/apiclient.py]: v3 merged code

**Review by Rrathinasabapath — APPROVED** [review]
Reviewed changes. They can be reviewed by Peter.

**Comment by arao6** [comment]
Madhan, the 'foreignExchanges' part need to be added to this.

**Review by arao6 — APPROVED** [review]

**Review by peter-joseph — APPROVED** [review]
